### PR TITLE
Models for Web UI

### LINKED ISSUE #8: Define data model
Setup data model the Web UI will use to render the widgets.

### PR DESCRIPTION
**Pull request by vova-beloded**
Fixes #8 

## Checklist

- [x] Documented

- [x] Unit tested

### CODE REVIEW

File: metrics/lib/features/dashboard/domain/usecases/get_build_metrics.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+class GetBuildMetrics implements UseCase<List<Build>, ProjectIdParam> {
+  final MetricsRepository _repository;
+
+  GetBuildMetrics(this._repository);
+
+  @override
+  Future<List<Build>> call(ProjectIdParam param) {
+    return _repository.getProjectBuilds(param.projectId);
+  }
+}
```
Comment (bootstraponline) [inline metrics/lib/features/dashboard/domain/usecases/get_build_metrics.dart]: Let's add newlines to the end of the files. Sometimes your IDE will have a setting to handle this automatically.

image

File: metrics/test/features/dashboard/presentation/state/project_metrics_store_test.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        actualFirstBuildStart.millisecondsSinceEpoch,
+        firstPerformancePoint.x,
+      );
+      expect(
+        actualFirstBuild.duration.inMilliseconds,
+        firstPerformancePoint.y,
+      );
+    });
+
+    test("Properly loads the build points", () async {
```
Comment (bootstraponline) [inline metrics/test/features/dashboard/presentation/state/project_metrics_store_test.dart]: These tests are awesome!

File: metrics/test/features/dashboard/presentation/state/project_metrics_store_test.dart [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      final actualFirstBuildDay = _trimToDay(actualFirstBuildStart);
+
+      final actualFirstBuildCount = actualBuildMetrics
+          .where((build) => _trimToDay(build.startedAt) == actualFirstBuildDay)
+          .length;
+
+      final actualAverageBuildTime = (actualBuildMetrics
```
Comment (illia-romanenko) [inline metrics/test/features/dashboard/presentation/state/project_metrics_store_test.dart]: Discussed possibility of moving more logic to the use case and making store more lean that takes care of presentation with Vova offline. Looking forward to the changes.

File: metrics/lib/features/dashboard/domain/entities/build_number_metric.dart [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/// Represents the [numberOfBuilds] during some [date].
```
Comment (illia-romanenko) [inline metrics/lib/features/dashboard/domain/entities/build_number_metric.dart]:
```suggestion
/// Represents the [numberOfBuilds] on specified [date].
```

File: metrics/lib/features/dashboard/domain/entities/coverage.dart [hunk 5]
```diff
@@ -10,4 +11,4 @@ class Coverage extends Equatable {
 
   @override
   List<Object> get props => [percent];
-}
+}
```
Comment (illia-romanenko) [inline metrics/lib/features/dashboard/domain/entities/coverage.dart]:
```suggestion
}

```

File: metrics/test/features/dashboard/domain/usecases/get_build_metrics_test.dart [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      final buildNumberMetrics = buildMetrics.buildNumberMetrics;
+      final firstBuildMetric = buildNumberMetrics.first;
+
+      expect(firstBuildMetric.date, buildStartDate);
+      expect(firstBuildMetric.numberOfBuilds, numberOfBuilds);
+    });
+  });
+}
+
+class MetricsRepositoryStabImpl implements MetricsRepository {
```
Comment (illia-romanenko) [inline metrics/test/features/dashboard/domain/usecases/get_build_metrics_test.dart]:
```suggestion
class MetricsRepositoryStubImpl implements MetricsRepository {
```

**Pull request merged**